code/modules/cargo/exports.dm

```diff
@@ -183,6 +197,51 @@ Credit dupes that require a lot of manual work shouldn't be removed, unless they
 
 GLOBAL_LIST_EMPTY(exports_list)
 
+/**
+  * Calculates the exact export value of the object based on the object's datum materials.
+  *
+  * loops through the atom's material datums and then multiplies that amount by it's relative worth, and sums the total.
+  */
+/datum/export/proc/get_material_cost(atom/movable/A)
+	for(var/i in A.custom_materials)
+		var/datum/material/M = i
+		. += M.value_per_unit * A.custom_materials[M]
+	. = round(.)  //floor of the value
+
+/**
+  * Sells an object using it's material cost.
+  *
+  * Works similarly to the sell object proc, but utilizes the get_material_cost proc in order to obtain it's raw value in terms of materials.
+  * Similarly checks for pricetags to split export profit the same way.
+  */
+/datum/export/proc/sell_material_item(obj/O, datum/export_report/report, dry_run = TRUE, apply_elastic = TRUE)
```

Comment:
```suggestion
/datum/export/proc/sell_material_item(obj/sold_object, datum/export_report/report, dry_run = TRUE, apply_elastic = TRUE)
```
Try to use more clear variables than upper case single letters. Also, is this proc used to sell items or objects?

Reply:
In this case it applies to objects on the whole, as both structures as well as objects can hold datum materials now. Also, everything else in the file used O as a precedent so I used obj/O for consistency sake.